src/wallet/spend.cpp

```diff
@@ -161,15 +161,17 @@ util::Result<PreSelectedInputs> FetchSelectedInputs(const CWallet& wallet, const
     PreSelectedInputs result;
     const bool can_grind_r = wallet.CanGrindR();
     for (const COutPoint& outpoint : coin_control.ListSelected()) {
-        int input_bytes = -1;
+        int32_t input_bytes = coin_control.GetInputWeight(outpoint).value_or(-1);
```

Comment:
(in commit 077783d24b7f1e3d3312e6cc12aee3f178400725)
Looking at the "override calculated size with coin control specified size" branch removed below, seems like a `GetVirtualTransactionSize` call is missing here (in the case that `GetInputWeight` doesn't return std::nullopt), as we want vbytes as `input_bytes` unit, not WU?

Reply:
Good catch, fixed.

Tried to add a test but the effects of this mistake are not readily apparent enough to be tested.